test/specs/rules/disallow-parentheses-around-arrow-param.js

```diff
@@ -29,4 +29,8 @@ describe('rules/disallow-parentheses-around-arrow-param', function() {
     it('should not report with parens around a multiple parameters with a default', function() {
         assert(checker.checkString('const a = (x = 1, y) => x * y;').isEmpty());
     });
+
+    it('should not report with use of destructuring #1672', function() {
```

Comment:
Please add a test for destructuring usage that's not the first param: `(foo, {bar}) => ` is valid.